e2e-tests/src/utils.rs

```diff
@@ -70,20 +89,127 @@ pub fn get_free_balance(account: &AccountId32, connection: &Connection) -> Balan
     connection.get_account_data(account).unwrap().unwrap().free
 }
 
+pub fn setup_for_transfer(config: Config) -> (Connection, AccountId32, AccountId32) {
+    let Config { node, seeds, .. } = config;
+
+    let (from, to) = get_first_two_accounts(&accounts(seeds));
+    let connection = create_connection(node).set_signer(from.clone());
+    let from = AccountId::from(from.public());
+    let to = AccountId::from(to.public());
+    (connection, from, to)
+}
+
 pub fn transfer(target: &AccountId32, value: u128, connection: &Connection) -> TransferTransaction {
-    let tx: UncheckedExtrinsicV4<_> = compose_extrinsic!(
+    crate::send_extrinsic!(
         connection,
         "Balances",
         "transfer",
+        |tx_hash| info!("[+] Transfer transaction hash: {}", tx_hash),
         GenericAddress::Id(target.clone()),
         Compact(value)
-    );
+    )
+}
+
+pub fn get_total_issuance(connection: &Connection) -> u128 {
+    connection
+        .get_storage_value("Balances", "TotalIssuance", None)
+        .unwrap()
+        .unwrap()
+}
+
+//////////////////////
+// Treasury related //
```

Comment:
Yeah, this is why `util` modules are a bad idea.

Generally the organization of code around here is a bit of a mess, including a cluttered `main.rs`. It would be great if you could split the code somewhat more readably. Possible approach – split them by component tested and just import the test functions into main (making it tiny and keeping the per-component code in one place). I'm not sure how much work that would be though, if you think it's too much for this PR at least do this for the treasury tests.

You might still end up with a `util` module (e.g. containing the function above this comment?), but hopefully much smaller and if everything goes well with a better name (e.g. `finance` for the above?). Kinda guessing here, 'cause I'm not intimately familiar with this code.

If you have a better idea as to how to keep this organized that would likely also be fine.

Reply:
I reorganized the package, but if it was just a transition between mess and mess', I can make further changes